### PR TITLE
chore: remove transform type coercion

### PR DESCRIPTION
**Pull request by lukesteensen**
This is just breaking some of the direct coupling that we have to the current `Transform` design, making it a bit easier to change moving forward.

### CODE REVIEW

File: src/transforms/metric_to_log.rs [hunk 1]
```diff
@@ -63,6 +63,17 @@ pub struct MetricToLogConfig {
     pub metric_tag_values: MetricTagValues,
 }
 
+impl MetricToLogConfig {
+    pub fn build(&self, context: &TransformContext) -> MetricToLog {
```
Comment (bruceg) [inline src/transforms/metric_to_log.rs:67]: I'm not excited about the overloaded `fn build` name here. I was confused about no longer needing the `.await` above and couldn't figure out how that worked until I saw this. Maybe call it `build_inner` or `build_sync` instead? I'm not hung up on what it's called, I just don't like the duplication.

**Review by bruceg — COMMENTED** [review]
I totally approve of this simplification, except for one gripe about a function name.

File: src/transforms/metric_to_log.rs (hunk 1 above)
Comment (lukesteensen) [reply src/transforms/metric_to_log.rs:67]: That is a very reasonable concern.

File: src/transforms/metric_to_log.rs [hunk 2]
```diff
@@ -63,6 +63,17 @@ pub struct MetricToLogConfig {
     pub metric_tag_values: MetricTagValues,
 }
 
+impl MetricToLogConfig {
+    pub fn build_transform(&self, context: &TransformContext) -> MetricToLog {
```
Comment (bruceg) [inline src/transforms/metric_to_log.rs:67]: Arguably the trait method should have been `build_transform`, but that's a much more invasive change.

**Review by bruceg — APPROVED** [review]

**Pull request merged**